### PR TITLE
update Undo/Redo to include addsc

### PR DESCRIPTION
**Pull request by Kokseng1**
undo/redo will update shortcut hashmaps, partially addressing PR 107 (Still missing view command and theme command)

Each `addressBookStateList` in `trackedAddressBook` contains a copy of all previous iterations `ShortcutSettings.`

Undo/Redo was originally implemented to only track commands that deal with `UniquePersonList persons`, hence every new command that requires Undo/Redo integration but does not actually edit the 'persons' list will be quite hard coded and prone to bugs. Please do share if anyone has a better idea for the integration of view and theme commands with undo/redo

### CODE REVIEW

**Comment by vansh284** [comment]
Haven't taken a look at the PR yet, but addressing this:
> Undo/Redo was originally implemented to only track commands that deal with `UniquePersonList persons`, hence every new command that requires Undo/Redo integration but does not actually edit the 'persons' list will be quite hard coded and prone to bugs.

Currently seems that we have a ` TrackedAddressBook`. Could this be replaced by a `TrackedModelManager` instead? Then we could encapsulate everything relevant for all the commands with this. Not sure if this would work, just an idea.

**Comment by Kokseng1** [comment]
> Currently seems that we have a `TrackedAddressBook`. Could this be replaced by a `TrackedModelManager` instead? Then we could encapsulate everything relevant for all the commands with this. Not sure if this would work, just an idea.

Sounds like a good idea, Ill try this for view and theme commands. Thanks!

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -37,13 +36,12 @@ public class LogicManager implements Logic {
...
     private final Storage storage;
     private final AddressBookParser addressBookParser;
-    private final CommandHistory commandHistory;
 
     /**
      * Constructs a {@code LogicManager} with the given {@code Model} and {@code Storage}.
      */
     public LogicManager(Model model, Storage storage) {
-        this.commandHistory = new CommandHistory();
+        //this.commandHistory = new CommandHistory();
```
Comment (vansh284) [inline src/main/java/seedu/address/logic/LogicManager.java:44]: Think we can delete this comment since its redundant now?

**Review by vansh284 — APPROVED** [review]
LGTM, just a small nit before merging.
As discussed in the comments, this should just be a temporay fix. We should replace the `TrackedAddressBook` with a `TrackedModelManager` in the future. Including the `ShortcutSettings` as a field under `AddressBook` seems counter-intuitive.

File: src/main/java/seedu/address/logic/LogicManager.java (hunk 1 above)
Comment (Kokseng1) [reply src/main/java/seedu/address/logic/LogicManager.java]: My bad for missing that out, I have removed it

**Pull request merged**